src/optimize.jl

```diff
@@ -486,11 +486,13 @@ method for use by [`build_optimizer_model!`](@ref).
 function clear_optimizer_model_build!(model::JuMP.Model)::JuMP.Model
```

Comment:
You could use https://jump.dev/JuMP.jl/stable/reference/models/#Base.empty!-Tuple{Model}.

You'd have to pre-cache the extension stuff before calling and re-add it after, but it should simplify accessing the internal fields.

Reply:
Using `empty!` nearly works, except I need to retain the chosen optimizer and any attributes it might have. Admittedly, this is a little hacky, but I am not aware of a better way.

Reply:
Upon closer inspection, I was able to make `empty!` work as needed.